src/screens/SendToken/SendTokenTransaction.js

```diff
@@ -177,76 +179,83 @@ class SendTokenTransaction extends React.Component<Props> {
         {!noRetry && (
           <ButtonWrapper>
-            <Button onPress={this.handleNavigationBack} title={t('button.retry')} />
+            <Button onPress={handleNavigationBack} title={t('button.retry')} />
           </ButtonWrapper>
         )}
-        <TouchableOpacity onPress={this.handleDismissal}>
+        <TouchableOpacity onPress={handleDismissal}>
           <CancelText>{t('button.cancel')}</CancelText>
         </TouchableOpacity>
-      </FailureButtonsWrapper>
+      </ButtonContainer>
     );
-  }
+  };
 
-  render() {
-    const { navigation } = this.props;
-    const {
-      isSuccess,
-      error,
-      transactionPayload: {
-        tokenType: transactionTokenType,
-        extra: {
-          allowance = {},
-        } = {},
-      },
-      transactionType,
-    } = navigation.state.params;
-
-    const animationSource = isSuccess ? animationSuccess : animationFailure;
-    const transactionStatusText = isSuccess
-      ? getTransactionSuccessMessage(transactionType)
-      : getTransactionErrorMessage(error);
-    const isAllowanceTransaction = Object.keys(allowance).length;
-    const transactionStatusTitle = isSuccess
-      ? getTransactionSuccessTitle({ transactionTokenType, transactionType, isAllowanceTransaction })
-      : t('error.transactionFailed.default');
-    const titleStyle = { ...objectFontStyles.large, marginTop: 16, marginBottom: 7 };
-    const textStyle = { ...objectFontStyles.regular, marginBottom: 75 };
-    return (
-      <Container>
-        <Wrapper flex={1} center regularPadding>
-          <Animation source={animationSource} />
-          <Title fullWidth title={transactionStatusTitle} align="center" noBlueDot titleStyles={titleStyle} noMargin />
-          <Paragraph light center style={textStyle}>{transactionStatusText}</Paragraph>
-          {isSuccess ? this.renderSuccessButton() : this.renderFailureButtons()}
-        </Wrapper>
-        {/*
+  const animationSource = isSuccess ? animationSuccess : animationFailure;
+  const transactionStatusText = isSuccess
+    ? getTransactionSuccessMessage(transactionType, chain)
+    : getTransactionErrorMessage(error);
+  const isAllowanceTransaction = Object.keys(allowance).length;
+  const transactionStatusTitle = isSuccess
+    ? getTransactionSuccessTitle({ transactionTokenType, transactionType, isAllowanceTransaction })
+    : t('error.transactionFailed.default');
+  const titleStyle = { ...objectFontStyles.large, marginTop: 16, marginBottom: 7 };
+  const textStyle = { ...objectFontStyles.regular, marginBottom: 75 };
+  return (
+    <Container>
+      <Wrapper flex={1} center regularPadding>
+        <Animation source={animationSource} />
+        <Title fullWidth title={transactionStatusTitle} align="center" noBlueDot titleStyles={titleStyle} noMargin />
+        <Paragraph light center style={textStyle}>
+          {transactionStatusText}
+        </Paragraph>
+        {isSuccess ? renderSuccessButton() : renderFailureButtons()}
```

Comment:
both success and failure have multiple buttons, maybe we can change this to just `renderSuccess` and `renderFailure`, wdyt?

Reply:
Or combine it into single and put back here combined and used with conditional rendering since both share nearly the same same structure

Reply:
@poocart Done changed to renderSuccess and renderFailure.